File: Relative_Ranks.py
score = [10,3,8,9,4]

def findRelativeRanks(score):
    s = score.copy()
    s.sort(reverse=True)
    out = []
    for i in s:
        if s.index(i) == 0:
            out.append("Gold Medal")
        elif s.index(i) == 1:
            out.append("Silver Medal")
        elif s.index(i) == 2:
            out.append("Bronze Medal")
        else:
            out.append(str(s.index(i)+1))
    return out


print(findRelativeRanks(score))

File: Relative_Ranks/Relative_Ranks.go
package main

import (
	"fmt"
	"sort"
	"strconv"
)

func main() {
	score := []int{10, 3, 8, 9, 4}
	fmt.Println(findRelativeRanks(score))
}
func findRelativeRanks(score []int) []string {
	s := make([]int, len(score))
	copy(s, score)
	sort.Sort(sort.Reverse(sort.IntSlice(s)))
	res := make([]string, len(score))
	mk := make(map[int]string)
	for i, j := range s {
		if i == 0 {
			mk[j] = "Gold Medal"
		} else if i == 1 {
			mk[j] = "Silver Medal"
		} else if i == 2 {
			mk[j] = "Bronze Medal"
		} else {
			mk[j] = strconv.Itoa(i + 1)
		}
	}
	for j, i := range score {
		fmt.Println(i)
		res[j] = mk[i]
	}
	return res
}
